aziotctl/aziotctl-common/src/config/super_config.rs

```diff
@@ -94,11 +94,17 @@ pub enum ProvisioningType {
         global_endpoint: Url,
         id_scope: String,
         attestation: DpsAttestationMethod,
+        #[serde(skip_serializing_if = "Option::is_none")]
+        payload: Option<Payload>,
     },
 
     /// Disables provisioning with IoT Hub for devices that use local identities only.
     None,
 }
+#[derive(Debug, Deserialize, Serialize)]
+pub struct Payload {
+    pub uri: String,
```

Comment:
Does url::Url work here instead of String?

Also, do we ever plan to support other formats for payload in the config?

Reply:
Yes, url::Url works. I've made the change and tested it out. See E2E tests here that were run with the change: https://github.com/nlcamp/iot-identity-service/actions/runs/2776588494

RE supporting other formats for payload, @micahl  and I have discussed possible future support for a `payload.value` parameter that would allow users to specify the json payload inline instead of using the uri parameter. This would be consistent with the way we support `device_id_pk = { uri = "..." }` or `device_id_pk = { value = "..." }`.